internal/connect/command_logs.go

```
@@ -0,0 +1,333 @@
+	connectorId := args[0]
+	if connectorId == "" {
+		return fmt.Errorf("connector ID cannot be empty")
+	}
+
+	startTime, err := cmd.Flags().GetString("start-time")
+	if err != nil {
+		return err
+	}
+
+	endTime, err := cmd.Flags().GetString("end-time")
+	if err != nil {
+		return err
+	}
+
+	level, err := cmd.Flags().GetString("level")
+	if err != nil {
+		return err
+	}
+	levels := strings.Split(level, "|")
+	for _, l := range levels {
+		if l != "INFO" && l != "WARN" && l != "ERROR" {
+			return fmt.Errorf("invalid log level: %s", l)
+		}
+	}
+
+	searchText, err := cmd.Flags().GetString("search-text")
+	if err != nil {
+		return err
+	}
+
+	outputFile, err := cmd.Flags().GetString("output-file")
+	if err != nil {
+		return err
+	}
+
+	next, err := cmd.Flags().GetBool("next")
+	if err != nil {
+		return err
+	}
+
+	if err := validateTimeFormat(startTime); err != nil {
+		return fmt.Errorf("invalid start-time format: %w", err)
+	}
+
+	if err := validateTimeFormat(endTime); err != nil {
+		return fmt.Errorf("invalid end-time format: %w", err)
+	}
+
+	if endTime < startTime {
+		return fmt.Errorf("end-time must be greater than start-time")
+	}
+
+	startTimeParsed, _ := time.Parse(time.RFC3339, startTime)
+	now := time.Now()
+	maxAge := 72 * time.Hour
+	if startTimeParsed.Before(now.Add(-maxAge)) {
+		return fmt.Errorf("start-time cannot be older than 72 hours as API can only return logs for the last 3 days")
+	}
+
+	currentLogQuery := &config.ConnectLogsQueryState{
+		StartTime:   startTime,
+		EndTime:     endTime,
+		Level:       level,
+		SearchText:  searchText,
+		ConnectorId: connectorId,
+		PageToken:   "",
+	}
+
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return fmt.Errorf("failed to get Kafka cluster information: %w\nPlease ensure you have set a cluster context with 'confluent kafka cluster use <cluster-id>' or specify --cluster flag", err)
+	}
+	kafkaClusterId := kafkaCluster.GetId()
```

Are we checking if this gives empty? Do we want to return an error if that is the case?

---

kafkaCluster.GetId() returns "" when kafkaCluster is nil and kafkaCluster is nil when the kafka.GetClusterForCommand returns error, so in short we wouldn't require any additional check here